lib/stages/RfiFrameDrop.cpp

```diff
@@ -0,0 +1,271 @@
+                         sk_exceeds[kk], std::get<1>(_thresholds[kk]));
+                    if (sk_exceeds[kk] > std::get<1>(_thresholds[kk])) {
+                        skip = true;
+                        _failing_frame_counter
+                            .labels({std::to_string(freq_id),
+                                     std::to_string(std::get<1>(_thresholds[kk])),
+                                     std::to_string(std::get<2>(_thresholds[kk]))})
+                            .inc();
+                    }
+                    // Reset counters for the next sub_frame
+                    sk_exceeds[kk] = 0;
+                }
+
+                // If no frame exceeded it's threshold then we should transfer the
+                // frame over to the output and release it. If we want to drop the
+                // incoming frame then we leave the output as is.
+                if (!skip || !enable_rfi_zero) {
+
+                    if (wait_for_empty_frame(_buf_out, unique_name.c_str(), frame_id_out)
+                        == nullptr) {
+                        break;
+                    }
+                    copy_frame(_buf_in_vis, frame_id_in_vis, _buf_out, frame_id_out);
+                    mark_frame_full(_buf_out, unique_name.c_str(), frame_id_out++);
+                } else {
+                    _dropped_frame_counter.labels({std::to_string(freq_id)}).inc();
+                }
+
+                mark_frame_empty(_buf_in_vis, unique_name.c_str(), frame_id_in_vis++);
+                _frame_counter.labels({std::to_string(freq_id)}).inc();
+            }
+        }
+        mark_frame_empty(_buf_in_sk, unique_name.c_str(), frame_id_in_sk++);
+    }
+}
+
+
+// mostly copied from visFrameView
+void RfiFrameDrop::copy_frame(Buffer* buf_src, int frame_id_src, Buffer* buf_dest,
+                              int frame_id_dest) {
+
+    // Buffer sizes must match exactly
+    if (buf_src->frame_size != buf_dest->frame_size) {
+        throw std::runtime_error(
+            fmt::format(fmt("Buffer sizes must match for direct copy (src {:d} != dest {:d})."),
+                        buf_src->frame_size, buf_dest->frame_size));
+    }
+
+    int num_consumers = get_num_consumers(buf_src);
+
+    // Copy or transfer the data part.
+    if (num_consumers == 1) {
+        // Transfer frame contents with directly...
+        swap_frames(buf_src, frame_id_src, buf_dest, frame_id_dest);
+    } else if (num_consumers > 1) {
+        // Copy the frame data over, leaving the source intact
+        std::memcpy(buf_dest->frames[frame_id_dest], buf_src->frames[frame_id_src],
+                    buf_src->frame_size);
+    }
+
+    pass_metadata(buf_src, frame_id_src, buf_dest, frame_id_dest);
+}
+
+bool RfiFrameDrop::rest_callback(nlohmann::json& update) {
+    bool enable_rfi_zero_new;
+    std::vector<std::tuple<float, size_t, float>> thresholds_new;
+
+    try {
+        enable_rfi_zero_new = update.at("enable_rfi_zero").get<bool>();
```

Comment:
This value is also in the chimeMetadata as the `rfi_zeroed` flag.  If you don't want us to also have to push updates to this endpoint at solar transit as well.   Unless you want to be able to control this independently.